src/features/applications/hooks/defaults.ts

```diff
@@ -14,6 +15,30 @@ hooks.addPureHook('POST', 'resin', 'application', {
 			);
 		}
 
+		request.values.uuid = request.values.uuid ?? uuid.v4().replace(/-/g, '');
+
+		// reconstitute the value into a properly formatted UUID...
+		const toUuid = (i: string): string => {
+			if (i.length !== 32) {
+				return '';
+			}
+
+			const parts: string[] = [];
+			parts.push(i.substr(0, 8));
```

Comment:
You could probably use `_.chunk(i, 4)`.
PS: And probably use a better param name than `i` ;)

Reply:
I didn't want to bring in lodash when a simple `substr` is ample.